### PR TITLE
Do some initial file syncing error handling cleanup

### PR DESCRIPTION
**Pull request by ericholscher**
This is rebased on top of the existing search cleanup, since it was touching similar code.

### CODE REVIEW

File: readthedocs/builds/syncers.py [hunk 1]
```diff
@@ -140,7 +140,8 @@ def copy(cls, path, target, host, is_file=False, **__):
         )
         ret = os.system(sync_cmd)
         if ret != 0:
-            log.info("COPY ERROR to app servers.")
+            log.error("COPY ERROR to app servers. Return code: {}".format(ret))
+            log.error("COPY ERROR command run: {}".format(sync_cmd))
```
Comment (agjohnson) [inline readthedocs/builds/syncers.py]: One log line would be better here, so we're not splitting reporting up into two line/pieces to sentry.

File: readthedocs/core/utils/__init__.py [hunk 2]
```diff
@@ -25,7 +26,12 @@
 SYNC_USER = getattr(settings, 'SYNC_USER', getpass.getuser())
 
 
-def broadcast(type, task, args, kwargs=None):  # pylint: disable=redefined-builtin
+def broadcast(type, task, args, kwargs=None, callback=None):  # pylint: disable=redefined-builtin
+    """
+    Run a broadcast across our servers.
+
+    Returns the task results for each task that was run.
```
Comment (agjohnson) [inline readthedocs/core/utils/__init__.py]: Probably want to explain `callback` usage here as well

File: readthedocs/core/utils/__init__.py [hunk 3]
```diff
@@ -34,12 +40,18 @@ def broadcast(type, task, args, kwargs=None):  # pylint: disable=redefined-built
...
-        task.apply_async(
-            queue=server,
-            args=args,
-            kwargs=kwargs,
-        )
+        task_sig = task.s(*args, **kwargs).set(queue=server)
+        tasks.append(task_sig)
+    if callback:
+        task_promise = chord(tasks)(callback).get()
+        log.debug('Sent Chord: {}'.format(locals()))
```
Comment (agjohnson) [inline readthedocs/core/utils/__init__.py]: If we can't guarantee that `locals()` won't leak things we don't want logged to logging, I'd say this should instead be logging some interpretation of the task name and what was passed in. Perhaps task name + kwarg names?

File: readthedocs/core/utils/__init__.py [hunk 4]
```diff
@@ -34,12 +40,18 @@ def broadcast(type, task, args, kwargs=None):  # pylint: disable=redefined-built
...
+    else:
+        task_promise = group(*tasks).apply_async()
+        log.debug('Sent Group: {}'.format(locals()))
```
Comment (agjohnson) [inline readthedocs/core/utils/__init__.py]: Same as above

File: readthedocs/settings/dev.py [hunk 5]
```diff
@@ -55,6 +56,8 @@ def DATABASES(self):  # noqa
     def LOGGING(self):  # noqa - avoid pep8 N802
         logging = super(CommunityDevSettings, self).LOGGING
         logging['formatters']['default']['format'] = '[%(asctime)s] ' + self.LOG_FORMAT
+        logging['handlers']['console']['level'] = 'DEBUG'
```
Comment (agjohnson) [inline readthedocs/settings/dev.py]: Do we want debug logs spamming the console? @humitos and I talked on this a while back and decided that moving DEBUG level logs to a file avoids overwhelming users. I'm -0 on moving back to the console.

File: readthedocs/settings/dev.py [hunk 6]
```diff
@@ -55,6 +56,8 @@ def DATABASES(self):  # noqa
...
+        logging['loggers']['']['handlers'] = []
```
Comment (agjohnson) [inline readthedocs/settings/dev.py]: Does this remove the double logging on development?

File: readthedocs/core/utils/__init__.py (hunk 3 above)
Comment (ericholscher) [reply readthedocs/core/utils/__init__.py]: I'll just kill this, as celery with debug logging will show the tasks in console

File: readthedocs/settings/dev.py (hunk 5 above)
Comment (ericholscher) [reply readthedocs/settings/dev.py]: OK. I found it easier to deal with, but that's fine as long as we know it is a convention. Is that what debug.log is for?

File: readthedocs/settings/dev.py (hunk 6 above)
Comment (ericholscher) [reply readthedocs/settings/dev.py]: Yea.

**Pull request merged**